core/domain/state_domain.py

```diff
@@ -110,11 +113,16 @@ def validate(self, interaction, exp_param_specs_dict):
                 'Expected answer group rules to be a list, received %s'
                 % self.rule_specs)
 
-        if self.tagged_misconception_id is not None:
-            if not isinstance(self.tagged_misconception_id, int):
+        if self.tagged_skill_misconception_id is not None:
+            if not isinstance(self.tagged_skill_misconception_id, basestring):
                 raise utils.ValidationError(
-                    'Expected tagged misconception id to be an int, '
-                    'received %s' % self.tagged_misconception_id)
+                    'Expected tagged skill misconception id to be a str, '
+                    'received %s' % self.tagged_skill_misconception_id)
+            if self.tagged_skill_misconception_id.find('-') == -1:
```

Comment:
Should also check that there's exactly one hyphen?

Also are hyphens possible in skill IDs or misconception IDs? If so, perhaps we should be using a different delimiter.

Reply:
Done. Hyphens are not possible in skillIDs or misconception IDs.